src/Proxy.NET/Services/CatalogService.cs

```diff
@@ -0,0 +1,109 @@
+        return result;
+    }
+
+    /// <summary>
+    /// Retrieves a catalog item.
+    /// </summary>
+    /// <param name="requestContext">The authorization response containing the event ID and deployment name.</param>
+    /// <returns>The deployment associated with the specified catalog name.</returns>
+    public async Task<Deployment> GetCatalogItemAsync(RequestContext requestContext)
+    {
+        var deployments = await GetDecryptedEventCatalogAsync(requestContext.EventId, requestContext.DeploymentName);
+
+        if (deployments.Count == 0)
+        {
+            throw new HttpRequestException(
+                $"The requested model '{requestContext.DeploymentName}' is not available for this event. "
+                    + $"Available models are: {string.Join(", ", (await GetEventCatalogAsync(requestContext.EventId)).Select(d => d.DeploymentName))}",
+                null,
+                HttpStatusCode.BadRequest
+            );
+        }
+
+        int index = new Random().Next(deployments.Count);
+        requestContext.CatalogId = deployments[index].CatalogId;
+
+        return deployments[index];
+    }
+
+    /// <summary>
+    /// Retrieves the capabilities for a given event.
+    /// </summary>
+    /// <param name="eventId">The ID of the event.</param>
+    /// <returns>A dictionary containing the capabilities for each deployment model type.</returns>
+    public async Task<Dictionary<string, List<string>>> GetCapabilities(string eventId)
+    {
+        var deployments = await GetEventCatalogAsync(eventId);
+        var capabilities = new Dictionary<string, List<string>>();
+
+        foreach (var deployment in deployments)
+        {
+            if (!capabilities.TryGetValue(deployment.ModelType, out var value))
+            {
+                value = [];
+                capabilities[deployment.ModelType] = value;
+            }
+            value.Add(deployment.DeploymentName);
+        }
+
+        return capabilities;
+    }
+
+    /// <summary>
+    /// Retrieves a list of deployments from the catalog for a specific event.
+    /// </summary>
+    /// <param name="event_id">The ID of the event.</param>
+    /// <returns>A task that represents the asynchronous operation. The task result contains a list of deployments.</returns>
+    private async Task<List<Deployment>> GetEventCatalogAsync(string event_id)
+    {
+        var result = await db
+            .OwnerCatalogs.Where(oc => oc.Active && oc.Events.Any(e => e.EventId == event_id))
+            .OrderBy(oc => oc.DeploymentName)
+            .Select(oc => new Deployment
+            {
+                DeploymentName = oc.DeploymentName,
+                ModelType = oc.ModelType.ToString() ?? "Type unknown",
+                Location = oc.Location
+            })
+            .ToListAsync(); // Fetch the data first
+
+        return result.DistinctBy(d => d.DeploymentName).ToList(); // Apply DistinctBy in memory
```

Comment:
Why wouldn't we have the `Distinct` call done on the DB?

Reply:
Because the EF complained the PG doesn't support Distinct. So I did in menory after the call...

Reply:
That sounds like a failure in the DB design 🤣